### DISCUSSION TITLE
CRC - port number `80` unavailable due to another process

### DISCUSSION THREAD

**Discussion by radaviko** [crc-org/crc#3323] [Q&A] [answered]
I installed RedHat OpenShift local on my Windows VDI.
C:\Program Files\Red Hat OpenShift Local>crc.exe start --log-level debug
DEBU CRC version: 2.7.1+8b30b973
DEBU OpenShift version: 4.11.0
DEBU Podman version: 4.1.1
DEBU Running 'crc start'

The install was successful but upon start I get error as RedHat OpenShift Local aka Code Ready Containers (CRC) has port 80 hardcoded in the install. 
It fails with error when I attempt to start the instance
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").EditionID'
failed to expose port :80 -> 192.168.127.2:80: listen tcp :80: bind: An attempt was made to access a socket in a way forbidden by its access permissions.

Windows OS already consuming the port 80 for system services. Is there any way to change or proxy the port for CRC to start without using port 80?

C:\Program Files\Red Hat OpenShift Local>netstat -ano | findstr 80
  TCP    0.0.0.0:80             0.0.0.0:0              LISTENING       4
  TCP    0.0.0.0:443            0.0.0.0:0              LISTENING       4800

Please advise of a solution as port 80/443 is used by Windows and I cannot start the CRC instance.

Thank you in advance.

**Comment by praveenkumar** [comment, marked as answer]
https://github.com/code-ready/crc/pull/3332 we are working to have an alternative port 8080/8443 in case 80/443 is in use, hopefully this will be available in upcoming release. you have to then make sure to use those alternative port to access the application routes.

**Reply by gbraad** [reply]
Related to #2760 and #3255

**Comment by radaviko** [comment]
Thanks for looking into the issue. Are there any updates please on the alternate port fix. Any additional details on the next release dates. Thank you very much in advance.

**Reply by praveenkumar** [reply]
@radaviko with latest release 2.9.0 we added config option for `ingress-https-port` and `ingress-http-port` which you can use to define different port than 443/80 so it shouldn't block you to run cluster but application route need to have those port info also.

like something following.
```
crc config set ingress-https-port 8443
crc config set ingress-http-port 8080
```

**Reply by ahsodex** [reply]
Doesn't work...

```
> crc config set ingress-http-port 8080
Changes to configuration property 'ingress-http-port' will break OpenShift HTTP routes.
In order to access OpenShift applications through HTTP URLs the 8080 port must be manually specified, such as http://myapp.apps-crc.testing:8080

> crc config set ingress-https-port 8443
Changes to configuration property 'ingress-https-port' will break OpenShift HTTPS routes.
In order to access OpenShift applications through HTTPS URLs the 8443 port must be manually specified, such as https://myapp.apps-crc.testing:8443
After this change, the OpenShift console will be non-functional because of OpenShift limitations

```

**Reply by praveenkumar** [reply]
@ahsodex what is not working? you hit something like https://github.com/crc-org/crc/issues/4124 one?

**Reply by ahsodex** [reply]
When I tried your solution, I got the output mentioned in my last post, so solution doesn't work.
Not sure about 4124, that is for Mac. I'm on Windows, having same problem as the OP for this issue (3323), using latest CRC / Podman Desktop. I tried searching the web for why ports are in use on Windows and lots of mention to disable http service(s), but I can't disable those as they are needed in my corporate environment. So CRC / Podman needs option to use different / non-used ports. Thanks.

**Reply by praveenkumar** [reply]
@ahsodex I think #4124 is for every platform, can you create an issue using issue template to figure out what actually breaking for you?
